src/sdam/server_selection.ts

```diff
@@ -31,6 +31,24 @@ export function writableServerSelector(): ServerSelector {
     );
 }
 
+/**
+ * The purpose of this selector is to select the same server, only
+ * if it is in a state that it can have commands sent to it.
+ */
+export function sameServerSelector(description?: ServerDescription): ServerSelector {
+  return (
+    topologyDescription: TopologyDescription,
+    servers: ServerDescription[]
+  ): ServerDescription[] => {
+    if (!description) return [];
+    // Filter the servers to match the provided description only if
+    // the type is not unknown.
+    return servers.filter((s: ServerDescription) => {
```

Comment:
```suggestion
    return servers.filter(sd => {
      return sd.address === description.address && sd.type !== ServerType.Unknown;
```

Mostly stylistic but our well typed stuff can make use of less annotations.